scipy/optimize/_linprog_util.py

```diff
@@ -1086,6 +1086,65 @@ def vstack(blocks):
     return A, b, c, c0, x0
 
 
+def _round_to_power_of_two(x):
+    """
+    Round elements of the array to the nearest power of two.
+    """
+    return 2**np.around(np.log2(x))
+
+
+def _autoscale(A, b, c, x0):
+    """
+    Scales the problem according to ___
+    """
+    m, n = A.shape
+
+    C = 1
+    R = 1
+
+    if A.size > 0:
+
+        R = np.max(np.abs(A), axis=1)
+        if sps.issparse(A):
+            R = R.toarray().flatten()
+        R[R == 0] = 1
```

Comment:
You might want to include for rows/cols whose max is less than some safe number to avoid overflow by reciprocation. See lapack's equilibriation routines for example [`?GEEQUB`](http://www.netlib.org/lapack/explore-html/dd/d9a/group__double_g_ecomputational_ga4547be7cf34938979c5d741d148e4d3b.html#ga4547be7cf34938979c5d741d148e4d3b)

Reply:
It looks like it just makes sure there are no scaling factors below `SMLNUM` or bigger than `BIGNUM`:
> R(i) and C(j) are restricted to be a power of the radix between SMLNUM = smallest safe number and BIGNUM = largest safe number.

If that is the case, is it as simple as:
```
smlnum = sys.float_info.min
bignum = 1/smlnum
R[R >= bignum] = bignum
R[R <= smlnum] = smlnum
# R[R == 0] = 1  # doesn't really need to be 1; just can't be 0. smlnum is OK.
```
and similar for `C`?

Do you plan to wrap it?

Reply:
I would but that's probably too late for your taste. I'm trying to fix my broken toolchain. I think 

```
R[np.abs(R) < np.spacing(1.)*100] = 1
```
would be safe enough for getting out of trouble.

Reply:
Yes, that certainly seems safe, but probably _too_ safe, no? The point of this routine is to avoid scaling issues, and this puts a tight restriction on the range of values it's useful for.

In [DGEEQUB](http://www.netlib.org/lapack/explore-html/dd/d9a/group__double_g_ecomputational_ga4547be7cf34938979c5d741d148e4d3b.html#ga4547be7cf34938979c5d741d148e4d3b), `smlnum` is [`dlamch( 'S' )`](https://netlib.sandia.gov/lapack/util/dlamch.f), which is [`tiny`](http://www.lahey.com/docs/lfpro78help/F95ARTINYFn.htm) of a double, which is "the smallest positive number of a numeric data type that can be represented without loss of precision." (Or it can be around the reciprocal of `huge` of a double, which is probably around the same order of magnitude.)

`np.spacing(1.)` is `2.220446049250313e-16` on my machine, whereas `sys.float_info.min` is `2.2250738585072014e-308`. If the point is just to prevent over/underflow when taking the inverse of `R`, using `np.spacing(1.)` as a lower limit seems too conservative.

Reply:
I think you can judge that better. I am not knowledgeable enough to guess what the typical values appearing in the coefficient matrices. However if these are going to be pushed to LAPACK routines at some point for any reason we are bounded by the machine precision of doubles and their reciprocals.

Reply:
@mdhaber let me run some tests either tomorrow or over the weekend and get back to you. 

@ilayn I am interested in being more active with the lower level side, if this wrapper isn't too urgent, perhaps I could make an attempt at creating it?

Reply:
#10741 😅